velox/common/memory/MemoryAllocator.h

```diff
@@ -378,12 +360,13 @@ class MemoryAllocator : public std::enable_shared_from_this<MemoryAllocator> {
       return size_;
     }
 
-    void setPool(MemoryPool* FOLLY_NONNULL pool) {
+    void setPool(MemoryPool* pool) {
```

Comment:
I wonder if it makes sense to remove this setter and pass the pool to the constructor. Would you document this method to explain how it is supposed to be used? Is it Ok to never set the pool?

Reply:
We can't set it in constructor as it is also used in memory allocator internally for temporal memory allocations which we don't have the owner pool. I previously thought the owner could either be a allocator or pool but it is a little bit complicated. The memory pool is the allocation owner of all external use cases and all the allocations go through memory pool and we set the owner there on success. I will put the description there.